tests/pdb_test.go

```diff
@@ -0,0 +1,134 @@
+
+	It("should ensure the Pod Disruption Budget (PDB) is deleted when the VM is stopped via guest OS", func() {
+		var err error
+
+		By("Creating test VM")
+		vm := &v1.VirtualMachine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testvm",
+				Namespace: testsuite.NamespaceTestDefault,
+			},
+			Spec: v1.VirtualMachineSpec{
+				RunStrategy: kvpointer.P(v1.RunStrategyManual),
+				Template: &v1.VirtualMachineInstanceTemplateSpec{
+					Spec: v1.VirtualMachineInstanceSpec{
+						EvictionStrategy: kvpointer.P(v1.EvictionStrategyLiveMigrate),
+						Domain: v1.DomainSpec{
+							Devices: v1.Devices{
+								Disks: []v1.Disk{
+									{
+										Name: "disk0",
+										DiskDevice: v1.DiskDevice{
+											Disk: &v1.DiskTarget{
+												Bus: "virtio",
+											},
+										},
+									},
+									{
+										Name: "cloudinitdisk",
+										DiskDevice: v1.DiskDevice{
+											Disk: &v1.DiskTarget{
+												Bus: "virtio",
+											},
+										},
+									},
+								},
+							},
+							Resources: v1.ResourceRequirements{
+								Requests: k8sv1.ResourceList{
+									k8sv1.ResourceMemory: resource.MustParse("128Mi"),
+								},
+							},
+						},
+						Volumes: []v1.Volume{
+							{
+								Name: "disk0",
+								VolumeSource: v1.VolumeSource{
+									ContainerDisk: &v1.ContainerDiskSource{
+										Image:           "registry:5000/kubevirt/cirros-container-disk-demo:devel",
+										ImagePullPolicy: k8sv1.PullIfNotPresent,
+									},
+								},
+							},
+							{
+								Name: "cloudinitdisk",
+								VolumeSource: v1.VolumeSource{
+									CloudInitNoCloud: &v1.CloudInitNoCloudSource{
+										UserDataBase64: "1234",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		By("Creating the VM")
+		_, err = kubevirt.Client().VirtualMachine(testsuite.NamespaceTestDefault).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Starting the VM")
+		err = kubevirt.Client().VirtualMachine(testsuite.NamespaceTestDefault).Start(context.Background(), vm.Name, &v1.StartOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Waiting for the VM to be ready")
+		Eventually(func() (*v1.VirtualMachineInstance, error) {
+			return kubevirt.Client().VirtualMachineInstance(testsuite.NamespaceTestDefault).Get(context.Background(), vm.Name, metav1.GetOptions{})
+		}, 180*time.Second, 1*time.Second).Should(BeAssignableToTypeOf(&v1.VirtualMachineInstance{}))
+
+		By("Verifying PDB for testvm existence")
+		Eventually(func(g Gomega) {
+			pdbs, err := kubevirt.Client().PolicyV1().PodDisruptionBudgets(testsuite.NamespaceTestDefault).List(context.Background(), metav1.ListOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(pdbs.Items).To(Not(BeEmpty()), "The Pod Disruption Budget should be created")
+		}, 60*time.Second, 1*time.Second).Should(Succeed())
```

Comment:
I'd prefer smth like this:
```suggestion
		Eventually(func() ([]policyv1.PodDisruptionBudget, error) {
			pdbs, err := kubevirt.Client().PolicyV1().PodDisruptionBudgets(testsuite.NamespaceTestDefault).List(context.Background(), metav1.ListOptions{})
			return pdbs.Items, err
		}, 60*time.Second, 1*time.Second).Should(Not(BeEmpty()), "The Pod Disruption Budget should be created")
```
Consider that ginkgo handles if the error returned is not nil.

Reply:
fixed